### PR TITLE
Add normal marking delete button functionality.

### PR DESCRIPTION
**Pull request by rodsouza9**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -201,7 +224,7 @@ class Numpad extends React.Component<NumpadProps, {}> {
         return (
             <div className="button-num-pad">
                 {this.renderNumButtons()}
-                <Button className= "button" variant="contained">DELETE</Button>
+                <Button onClick={this.props.onClickDel} className= "button" variant="contained">DELETE</Button>
```
Comment (advait) [inline src/App.tsx]: Fix whitespace

File: src/App.tsx [hunk 2]
```diff
@@ -78,6 +78,25 @@ class Board extends React.Component<BoardProps, BoardState> {
         }
     }
 
+    deleteSelectedSquares() : void {
+        for (let i = 0; i < this.state.highlights.length; i++) {
+            let addy : SquareAddress = this.state.highlights[i];
```
Comment (advait) [inline src/App.tsx]: Rename to address

File: src/App.tsx [hunk 3]
```diff
@@ -78,6 +78,25 @@ class Board extends React.Component<BoardProps, BoardState> {
         }
     }
 
+    deleteSelectedSquares() : void {
+        for (let i = 0; i < this.state.highlights.length; i++) {
```
Comment (advait) [inline src/App.tsx]: Use `for of` here

File: src/App.tsx [hunk 4]
```diff
@@ -78,6 +78,25 @@ class Board extends React.Component<BoardProps, BoardState> {
...
+            let addy : SquareAddress = this.state.highlights[i];
+            if (!this.state.permanentValues.includes(addy)) {
+                let newValues = JSON.parse(JSON.stringify(this.state.values));
+                if (newValues[addy]) {
+                    newValues[addy] = null;
+                }
+                /**
+                  * //TODO: CREATE marking state and transfer marking logic to board state.
+                  */
+                this.setState({
```
Comment (advait) [inline src/App.tsx]: Couple issues:
1. Your logic is incorrect
2. Never call setstate in a loop. Clone the state, make all modifications to the clone, then call set state once at the end.

File: src/App.tsx [hunk 5]
```diff
@@ -158,7 +180,7 @@ class Square extends React.Component<SquareProps, {}> {
     render() {
         let light : string = this.props.isHighlighted ? "highlight" : (this.props.isContradicting ? "contradict" : "");
         let marks : boolean[] = this.getMarks();
-        if (this.props.markings && this.props.markings.length &&!this.props.isPermanent) {
+        if (this.props.markings && this.props.markings.length &&!this.props.isPermanent && !this.props.value) {
```
Comment (advait) [inline src/App.tsx]: 1. Fix whitespace
2. This logic is too confusing to me. Please either add a comment or add a helper function that makes it more clear why you need so many conditions in this if

File: src/App.tsx [hunk 6]
```diff
@@ -78,6 +78,25 @@ class Board extends React.Component<BoardProps, BoardState> {
         }
     }
 
+    deleteSelectedSquares() : void {
+        for (let i = 0; i < this.state.highlights.length; i++) {
+            let addy : SquareAddress = this.state.highlights[i];
+            if (!this.state.permanentValues.includes(addy)) {
+                let newValues = JSON.parse(JSON.stringify(this.state.values));
+                if (newValues[addy]) {
```
Comment (advait) [inline src/App.tsx]: This if is unnecessary

**Review by advait — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 6 above)
Comment (rodsouza9) [reply src/App.tsx]: its to make sure that you don't delete permanent values. You don't want to be able to delete the values passed into the board.

Comment (rodsouza9) [reply src/App.tsx]: oh i see

**Pull request merged**